linux/devdoc/threadpool_linux_requirements.md

```diff
@@ -0,0 +1,258 @@
+# threadpool_linux
+
+## Overview
+
+`threadpool_linux` is the Linux-specific implementation of the `threadpool` module.
+
+## Design
+
+`threadpool_linux` provides the Linux implementation of `threadpool`. The `threadpool` object contains an array of threads all initialized in `threadpool_open_async`  and an array of tasks all initialized in `threadpool_create`, all incoming tasks are being added in `threadpool_schedule_work` and scheduled in `threadpool_work_func`. 
+
+`threadpool` maintains the following states:
+1. `state`: a 32-bit variable represents a state enum value, which contains: 
+    - `THREADPOOL_STATE_NOT_OPEN` : entered after a call to `threadpool_create` or a call to `internal_close`.
```

Comment:
>entered after a call to `threadpool_create` or a call to `internal_close`. [](http://example.com/codeflow?start=36&length=74)

I guess these states will actually be removed from here when we say we will use sm.
But generally (for other times when we describe states) we should explain what the state means first. For example 
"OPEN" means the module can accept execute calls.
"CLOSING" means the module is performing close actions and all existing API calls will be drained.

Then, optionally next to the state you can specify when we enter that state.

Reply:
removed